src/i2p.cpp

```diff
@@ -202,7 +202,11 @@ bool Session::Accept(Connection& conn)
         return true;
     }
 
-    Log("Error accepting%s: %s", disconnect ? " (will close the session)" : "", errmsg);
+    if (m_interrupt) {
```

Comment:
Should this not be `*m_interrupt`?

Reply:
Good catch! The implicit cast to `bool` of `CThreadInterrupt` makes it easy to get such mistakes if the variable is a pointer to `CThreadInterrupt` or `std::optional<CThreadInterrupt>`. If `CThreadInterrupt` had an explicit method like `IsInterrupted()` then it would be more clear to call that one, but unfortunately it does not have such a method.

To be explicit we can do `if (m_interrupt->operator bool()) {` :laughing: :robot: :rofl: 

Or, more seriously, because `m_interrupt` can never be nullptr, change it to a reference.